### ISSUE TITLE
Add howto-style documentation for Twisted Web support of WSGI

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3510]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3510|
|Type|enhancement|
|Created|2008-10-28 19:29:28Z|

The [doc/web/howto/using-twistedweb.xhtml using twisted web](doc/web/howto/using-twistedweb.xhtml using twisted web) howto doesn't cover WSGI.  People are probably at least as interested in this one feature as they are in all the other features of twisted web combined.  There should be some docs about it.

Attachments:

* [t.w.wsgi-doc.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ae1/ae168e6ae4816a7e68163c9fc9575bc779e6bb32/ef8c821c6b15c895544f6c1a43000f6924b07ef1.diff) (1948 bytes) - added by ali on 2009-01-02 22:03:53Z - Patch to twisted web howto to illustrate wsgi
* [t.w.wsgi-doc2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ae1/ae168e6ae4816a7e68163c9fc9575bc779e6bb32/c39beede2e7d857c4d92e4d56cc991a0f5f05bc6.diff) (2974 bytes) - added by ali on 2009-03-06 17:00:22Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3510 3510
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation documentation
time__1225222168000000 1225222168000000
changetime__1236978958000000 1236978958000000
version__None None
owner__ 
cc__aafshar@... cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ali's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|ali commented|
|-|-|

I have made a very small description and example for doing this. Please comment on what needs expanding, contracting.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to ali|
|-|-|

  1. Names should conform to the Twisted coding standard.  Not `foo_bar`, but `fooBar`.
  1. `<pre language="python">` doesn't look right to me.  I think you meant `<pre class="python>`.  Did some lore docs somewhere give you the idea to use `language`?
  1. Most of the example code in this document are presented as .tac files (though not all).  I think it would be good to present this one as a .tac file, too.
  1. _non asynchronous_ is essentially a double negative.  _synchronous_ will do just as well. :)
  1. If you use `<code class="api">` instead of just `<code>`, links will be generated to the relevant API documentation.
  1. It might be good to mention that `twistd web --wsgi ...` can also be used to run an existing wsgi application if further customization isn't necessary (this approach won't let you serve up anything except the wsgi application, which might not be what you always want).  Since this is simpler, it probably makes sense to talk about it first, then move on to the more complicated way of setting up the `WSGIResource` directly.

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="ali's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|ali commented|
|-|-|

Thanks for the review. I think I have fixed all the items mentioned. It might be nice to have a separate document though, with the rise of WSGI 'n all.

**Comment by twisted-trac** [comment]
|<img alt="ali's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ali set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

Looks good.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [26366]) Apply t.w.wsgi-doc2.diff

Author: ali
Reviewer: exarkun
Fixes: [#3510](https://github.com/twisted/twisted/issues/3510)

Add howto-style documentation for Twisted Web's WSGI support.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
